### PR TITLE
move jws to k8s.io/cluster-bootstrap

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**: ref kubernetes/kubeadm#1600

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

/cc @rosti 
/assign @liggitt @neolit123 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
@yastij 
i think a better home for the JWS utils is the cluster-bootstrap repository.
https://github.com/kubernetes/cluster-bootstrap

to be able to push there you need to modify files under:
https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/cluster-bootstrap/

possibly under `/jws/...` under the root path.
but no strong opinions on my side.

EDIT: also this was used by the controller-manager.

**Comment by neolit123** [comment]
the cluster-boostrap home for this SGTM.
comments from @kubernetes/sig-cluster-lifecycle-pr-reviews ?

**Comment by dixudx** [comment]
> the cluster-boostrap home for this SGTM.

@neolit123 `k8s.io/cluster-bootstrap/util/jws` ?

**Comment by neolit123** [comment]
> k8s.io/cluster-bootstrap/util/jws

i didn't recommend util because there is already https://github.com/kubernetes/cluster-bootstrap/tree/master/token/util

so in a way token utils fall under token/util. following the existing pattern it should be jws/util (because the files we are moving a JWS related utils).

but yeah, no strong opinions on the naming.

**Comment by dixudx** [comment]
> i didn't recommend util because there is already https://github.com/kubernetes/cluster-bootstrap/tree/master/token/util

> so in a way token utils fall under token/util.

@neolit123  IIRC, we would make a move to `util/xxx`.

**Comment by yastij** [comment]
@dixudx - we can leave it under jws/util for now, and refactor it with token later.

**Comment by neolit123** [comment]
> @dixudx - we can leave it under jws/util for now, and refactor it with token later.

ok lets do this for now.

then we can do this:
> @neolit123 IIRC, we would make a move to util/xxx.

but we might break existing token users, if any.

**Comment by dixudx** [comment]
@yastij Yeah. Move on for now.

> but we might break existing token users, if any.

@neolit123  No worries. CI would test us.

File: staging/src/k8s.io/cluster-bootstrap/jws/jws.go [hunk 1]
```diff
@@ -23,10 +23,10 @@ import (
 	jose "gopkg.in/square/go-jose.v2"
 )
 
-// computeDetachedSig takes content and token details and computes a detached
+// ComputeDetachedSig takes content and token details and computes a detached
 // JWS signature.  This is described in Appendix F of RFC 7515.  Basically, this
 // is a regular JWS with the content part of the signature elided.
-func computeDetachedSig(content, tokenID, tokenSecret string) (string, error) {
+func ComputeDetachedSig(content, tokenID, tokenSecret string) (string, error) {
```
Comment (liggitt) [inline staging/src/k8s.io/cluster-bootstrap/jws/jws.go]: if we're exporting this, name it `ComputeDetachedSignature`

**Comment by liggitt** [comment]
one comment on the exported method name. will defer to @neolit123 on package structure within the cluster-bootstrap package. reassign me when this is ready for approval

/unassign

File: staging/src/k8s.io/cluster-bootstrap/jws/jws.go (hunk 1 above)
Comment (rosti) [reply staging/src/k8s.io/cluster-bootstrap/jws/jws.go]: +1 on the name change

**Review by rosti — COMMENTED** [review]
Thanks @yastij !
I think, that `k8s.io/cluster-bootstrap/token/jws` is a fine place for this.

**Review by luxas — APPROVED** [review]
I agree with the other comments, e.g. move to `k8s.io/cluster-bootstrap/token/jws` and the name change, but overall this move looks clean to me

/approve

Others can LGTM later

**Comment by yastij** [comment]
@neolit123 - this should be ready to merge

**Review by neolit123 — COMMENTED** [review]
/lgtm
thanks

**Pull request merged**